lib/gruff/base.rb

```diff
@@ -670,6 +674,12 @@ def draw_unique_label(index)
       end
     end
 
+    def hide_labels?
+      return @hide_labels if is_a? Gruff::Bar
```

Comment:
Each graph type handles labels differently and can set @hide_line_markers within their initializer.  This complicates the transition to handling hiding of labels and hiding of line markers separately.

Using the hide_labels? method allows for a transition of one graph type at a time, keeping the required changes simple and easy to review.

Reply:
I think we need to consider a SideBar too in a condition at least. What do you think about ?

Reply:
If we were to adapt this code to Gruff's style, it would look something like below.

```diff
diff --git a/lib/gruff/bar.rb b/lib/gruff/bar.rb
index b27f02e..2349e87 100644
--- a/lib/gruff/bar.rb
+++ b/lib/gruff/bar.rb
@@ -33,6 +33,10 @@ class Gruff::Bar < Gruff::Base
   # Default is +false+.
   attr_writer :show_labels_for_bar_values
 
+  # Prevent drawing of column labels below a bar graph. For other graph types, hide_labels
+  # is ignored and continues to be tied to hide_line_markers. Default is +false+.
+  attr_writer :hide_labels
+
   def initialize_ivars
     super
     @spacing_factor = 0.9
@@ -136,4 +140,8 @@ def draw_bars
   def calculate_spacing
     @scale * @group_spacing * (column_count - 1)
   end
+
+  def hide_labels?
+    @hide_labels
+  end
 end
diff --git a/lib/gruff/base.rb b/lib/gruff/base.rb
index c9fb7a7..979ca73 100644
--- a/lib/gruff/base.rb
+++ b/lib/gruff/base.rb
@@ -103,10 +103,6 @@ class Base
     # Prevent drawing of line markers. Default is +false+.
     attr_writer :hide_line_markers
 
-    # Prevent drawing of column labels below a bar graph. For other graph types, hide_labels
-    # is ignored and continues to be tied to hide_line_markers. Default is +false+.
-    attr_writer :hide_labels
-
     # Prevent drawing of the legend. Default is +false+.
     attr_writer :hide_legend
 
@@ -675,8 +671,6 @@ def draw_unique_label(index)
     end
 
     def hide_labels?
-      return @hide_labels if is_a? Gruff::Bar
-
       @hide_line_markers
     end
 
```